copy.go

```diff
@@ -171,7 +171,18 @@ func dcopy(srcdir, destdir string, info os.FileInfo, opt Options) (err error) {
 func onsymlink(src, dest string, opt Options) error {
 	switch opt.OnSymlink(src) {
 	case Shallow:
-		return lcopy(src, dest)
+		err := lcopy(src, dest)
+		if err != nil {
+			return err
+		}
+		if opt.PreserveTimes {
+			info, err := os.Lstat(src)
```

Comment:
Getting Lstat should be in `preserveLstat`. Wdyt?

Reply:
Agreed. My initial thinking was to keep the non-OS specifics outside of `preserveLtimes()`, but I totally see the redundancy argument for unsupported architectures. Since right now it's just binary supported / unsupported I'll move it over. Thx for the feedback!

Reply:
Done. Also made the call `unix`-specific to reduce some (now) unnecessary conversions and added a negative test to ensure any error is guaranteed to be caught by `os.IsNotExist()`.